### DISCUSSION TITLE
Custom Events for Embedded

### DISCUSSION THREAD

**Discussion by Lucien950** [slint-ui/slint#5453] [Q&A]
I am currently developing an embedded application which needs to listen to GPIO switch events similar to keyboard events. I am not sure how to implement this, while considering best practices concerning focus.

**Comment by ogoffart** [comment]
The way to do that is to forward them to the main event loop.
Assuming you get an interrupt that change some static atomic or something. Then you can query these value at every iteration of the loop.
For example https://github.com/ogoffart/carousel-mxchip-iot-devkit/blob/master/src/main.rs#L158-L170
